trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/signing/ChunkSigningSession.java

```diff
@@ -22,7 +22,7 @@
 import static java.util.Objects.requireNonNull;
 
 @SuppressWarnings("UnstableApiUsage")
-class ChunkSigningSession
+public class ChunkSigningSession
```

Comment:
This is the only part that needs to be `public`. Also, we could consider making `ChunkSigningSession` an interface and having a package-private implementation.